packages/flutter/lib/src/painting/decoration_image.dart

```diff
@@ -431,6 +435,29 @@ void paintImage({
     // as we apply a nine-patch stretch.
     assert(sourceSize == inputSize, 'centerSlice was used with a BoxFit that does not guarantee that the image is fully visible.');
   }
+
+  // Output size is fully calculated.
+  assert(() {
+    if (debugImageOverheadAllowedInKilobytes != null) {
+      final int acutalSizeKilobytes = (image.width * image.height * 4 * (4/3)) ~/ 1024;
```

Comment:
nit: why not take the delta in bytes and only round to kb on display? As written this could introduce some slightly strange rounding errors where warnings would be issued when the size difference was actually slightly less than the threshold.

Reply:
I've shifted this around to avoid some potential rounding issues. I still want the comparison to be in KB because bytes are too small, and my thought is that developers shouldn't be concerned with deltas < 1kb.